src/linear_algebra/matrix/symmetric.lean

```diff
@@ -0,0 +1,115 @@
+/-
+Copyright (c) 2021 Lu-Ming Zhang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Lu-Ming Zhang
+-/
+import data.matrix.block
+
+/-!
+# Symmetric matrices
+
+This file contains the definition and basic results about symmetric matrices.
+
+## Main definition
+
+ * `matrix.is_sym `: a matrix `A : matrix n n α` is "symmetric" if `Aᵀ = A`.
+
+## Tags
+
+sym, symmetric, matrix
+-/
+
+variables {α n m R : Type*}
+
+namespace matrix
+
+open_locale matrix
+
+/-- A matrix `A : matrix n n α` is "symmetric" if `Aᵀ = A`. -/
+def is_sym (A : matrix n n α) : Prop := Aᵀ = A
```

Comment:
I think this would be better as `is_symm`, since we use `.symm` as a spelling all the time.

Reply:
If you think `symm` is better, I can change to `symm` in this file.

Reply:
I've made a poll [here](https://leanprover.zulipchat.com/#narrow/stream/144837-PR-reviews/topic/.238955.20matrix.2Eis_sym.20or.20matrix.2Eis_symm/near/251708564)

Reply:
According to the poll results, I will change the name.